edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter.Test/ConnectionEventTest.cs

```diff
@@ -121,6 +122,43 @@ void Spy(RpcPacket packet)
             }
         }
 
+        [Fact]
+        public async Task WhenNoConnectionMessagesDontGetDropped()
```

Comment:
I think the test should just check that it throws IotHubException when it is disconnected. 
There should be an integration test in CloudEndpointTest to check that the message is not dropped.